### DISCUSSION TITLE
Type > into the main menu to open term - not working?

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#756] [Errors] [answered]
This was working in prior releases. Noticed that in 1.8.1 if I bring up the app and Type > into the main menu, the preview window flashes to different content for a tiny instant (maybe that was a terminal window going by?) but then immediately returns to the prior app state (showing me a list of scrips I could run). I don't think I've got conflicts with anything else trapping that keystroke...

**Comment by johnlindquist** [comment]
Is there anything in the ~/.kit/logs/kit.log ?

(The feature should still be working)

**Reply by cfritze64** [reply]
[2022-04-12 13:29:23.869] [info] 🏚  main shortcut
[2022-04-12 13:29:23.870] [info] 🏃‍♀️ Run /Users/BikeFixe/.kit/main/index.js
[2022-04-12 13:29:23.870] [info] Found: 
[2022-04-12 13:29:23.872] [info] 89172: 🏎 /Users/BikeFixe/.kit/main/index.js 
[2022-04-12 13:29:23.876] [info] 91197: 🟢 start idle Prompt
[2022-04-12 13:29:23.946] [info] ↖ OPEN: { x: 336, y: 134, width: 768, height: 480 }
[2022-04-12 13:29:30.779] [info] 89172: 🟡 exit 0. Prompt process: /Users/BikeFixe/.kit/main/index.js
[2022-04-12 13:29:30.780] [info] 89172: 🛑 removed

**Reply by johnlindquist** [reply]
Hmm, nothing there...

I'll be back at my computer in an hour and investigate some more ideas

**Reply by cfritze64** [reply]
no rush. I'm loving all the new single letter mode shortcuts 

An odd thing: if I search within .kit or .kenv folders with Atom for "term(" I don't find a file that has an obvious definition for the term global...

**Reply by johnlindquist** [reply]
It should be in `~/.kit/api/pro.js`:

https://github.com/johnlindquist/kit/blob/main/src/api/pro.ts#L282

(pro was originally meant for "paid" features, but I still haven't decided on them yet)

I always search for `term =` to hop there quickly.

**Reply by johnlindquist** [reply]
Can you run this in the terminal:

```bash
~/.kit/node/bin/node --version
```

Let me know what version of node your Script Kit install has?

**Reply by cfritze64** [reply]
cfritze macbook air:~ BikeFixe$ ~/.kit/node/bin/node --version
v15.12.0
cfritze macbook air:~ BikeFixe$ 
cfritze macbook air:~ BikeFixe$ node -v
v16.13.1

**Reply by johnlindquist** [reply]
Sorry, it needs to be the full path `~/.kit/node/bin/node --version`, not just `node`. `node` will default to your System's node, not Kit's.

**Reply by cfritze64** [reply]
I sent too much info at once. I did Kit's version too up there ^^
it's 15.12.0

**Reply by johnlindquist** [reply]
Could you try:
1. Quit kit
2. Delete the `~/.kit` dir
3. Open kit

Kit will then re-install the `.kit` with the latest bundled node. It should be v17.2

**Reply by cfritze64** [reply]
I gotta leave for a long hour. Your instructions did result in a fresh re-install of the bits. Version is 17.2 now, but the orig behavior persists... :-(

**Reply by johnlindquist** [reply]
In the file:

`~/.kit/run/pty.js`

On line 100, can you try changing:

```js
port = process.env.PORT || (await getPort({ port: 3131 }));
```

to

```js
port = 3132 //or any other number
```

Then hit the `cmd+;` shortcut ~3 times so that Kit opens/closes 3 times.

## Why 3 times?
Kit has a "process pool" of 3 idle processes that load in the SDK. The pool needs to be cleared before it will pick up a "hot swap" change of code from the SDK like we're doing above.

**Reply by cfritze64** [reply]
OK. Made that change and cleared the process pool. Even tried quitting+restarting the app as well, but no joy.

I did notice by watching real close: when I hit shift-. (greater than) the thing that flashes by is Greater than - or something like that - indicating I think the greater than handler, but before you can even read all of it, we're back to main.

also, a script with just "await term()" gives nothing.

**Reply by cfritze64** [reply]
(pty.js line 100 is called from the main window when I type greater than) Many apologies if I sent you down a rabbit hole before deleting my prior comment. I'll leave the debugging to the experts

**Comment by cfritze64** [comment, marked as answer]
I inadvertently broke the path to python on my system. Resolved. thx John!
